Initial PR

HarshNarayanJha opened the pull request:
Initial PR

Closes #2

HarshNarayanJha committed:
Initial Commit

HarshNarayanJha committed:
feat: list and launch zed with selected workspace

HarshNarayanJha committed:
feat: add initial macos support

HarshNarayanJha committed:
feat: add more linux binaries

HarshNarayanJha committed:
feat: option to use icon from system and zed preview support

HarshNarayanJha committed:
v0.2

HarshNarayanJha committed:
add md_lib_dependencies

HarshNarayanJha committed:
update plugin to plugin iid v3.0

HarshNarayanJha committed:
update year in __init__.py

HarshNarayanJha committed:
updates for repo transfer to @albertlauncher org

ManuelSchneid3r commented:
If there are no svgs available I'd say 128x128. Icons are 40x40 iirc. I.e. hdpi screens need at least 80x80

HarshNarayanJha committed:
resize icons to 128x128

ManuelSchneid3r commented:
Sorry for the late response

HarshNarayanJha commented:
Sorry my bad, I am using `xdg:zed` and `xdg:zed-preview`, and they do work. It's been a long time, that's why I don't remember, I have a toggle to use system icons which uses `xdg:zed`, which comes from the icon pack (on Linux at least).
By default, it uses the official zed icon, shipped with the plugin.

I think I will remove the icons entirely, it was my personal preference after all.

ManuelSchneid3r commented:
For the shipped official plugin we should respect user preferences. the xdg icon lookup chooses the official icon if no theme icon is available. Users can also overwrite this icon. I don't remember how as I never did, but I remember the icon lookup spec offers this opportunity. You could use this for your personal preference. 
If the plugin should work without having the app installed we need a fallback. The icon urls fields is an array for this purpose.

ManuelSchneid3r commented:
For jetbrains we should do this as well because they chnage quite often

HarshNarayanJha commented:
> For the shipped official plugin we should respect user preferences. the xdg icon lookup chooses the official icon if no theme icon is available. Users can also overwrite this icon. I don't remember how as I never did, but I remember the icon lookup spec offers this opportunity. You could use this for your personal preference.

Yeah, I think dropping the icon in `~/.local/share/icons` or ~/.icons` should work.

> If the plugin should work without having the app installed we need a fallback. The icon urls fields is an array for this purpose.

Got it. But this is not needed for this plugin, since zed needs to be installed.

HarshNarayanJha committed:
remove systemicon option. plugin now uses system icons to respect user's theming.

- remove icons from plugin source

HarshNarayanJha committed:
docs: add note for installing zed cli

HarshNarayanJha committed:
change to local_paths (blob) to local_paths_array (text)

- note that although this field helps get all workspaces, but it might contain multiple paths which I don't have

HarshNarayanJha committed:
v1.1

HarshNarayanJha committed:
update workspace path column name

latest zed version renamed column `paths`

NOTE: users will need the latest version of zed, otherwise workspaces won't load

HarshNarayanJha committed:
v2.0

- adhere to albert 33.0.0 - python interface version 4.0 - new config option `match_path`

HarshNarayanJha committed:
minor bugfix for config

HarshNarayanJha committed:
v3.0

- updates for albert 34.0.1 - python interface version 5.0
- add two new actions for opening terminal and file explorer at workspace path
- bug fixes and async item generation thanks to new albert API

HarshNarayanJha committed:
revert installation path change in readme
